CUMULUS-3847: Remove any remaining ES indexing functions and tests

charleshuang80 opened the pull request:
…loud Metrics usage

**Summary:** Summary of changes

Addresses [CUMULUS-3847: Remove any remaining ES indexing functions and tests](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3847)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

jennyhliu commented on packages/api/lambdas/cleanExecutions.js:
Suggest to comment out the code, and put a dummy handler, so the lambda can still be created in tf-modules/archive/clean_executions.tf

charleshuang80 replied:
code was removed, dummy handler added, as per other thread in this file

jennyhliu commented on packages/api/lambdas/bootstrap.js:
```diff
@@ -4,6 +4,7 @@
  * It helps:
  *  - adding ElasticSearch index mapping when a new index is created
  */
+// is this needed for CloudMetrics?
```
With es instance going away, this lambda shouldn't be needed.

jennyhliu commented on packages/aws-client/tests/test-services.js:
```diff
@@ -188,6 +188,7 @@ test('ec2() service defaults to localstack in test mode', async (t) => {
   );
 });
 
+// can this test be removed or is it needed for the granule CloudMetrics queries?
```
With es instance going away, I think packages/aws-client/tests/services.js es and this test are not needed.

charleshuang80 replied:
when I tried removing ES, I found that some of the ES vars were for Metrics, so it seemed like ES needs to stay for these tests

charleshuang80 replied:
ES references removed

jennyhliu commented on packages/api/lambdas/create-reconciliation-report.js:
```diff
@@ -969,6 +954,7 @@ async function handler(event) {
   process.env.CMR_PAGE_SIZE = process.env.CMR_PAGE_SIZE || '200';
 
   //TODO: Remove irrelevant env vars from terraform after ES reports are removed
+  // can this be changed now?
```
Yeah, we are not querying es anymore.

jennyhliu commented on packages/api/tests/lambdas/test-bulk-operation.js:
```diff
@@ -49,7 +49,7 @@ const esSearchStub = sandbox.stub();
 FakeEsClient.prototype.scroll = esScrollStub;
 FakeEsClient.prototype.search = esSearchStub;
-
+// is this es-client stuff still needed b/c granules and metrics?
```
I think es-client is still needed. You can try to take it out.

charleshuang80 replied:
when I tried removing ES, I found that some of the ES vars were for Metrics, so it seemed like ES needs to stay for these tests

npauzenga commented on CHANGELOG.md:
```diff
@@ -92,6 +92,8 @@ External tooling making use of `searchContext` in the `GET` `/granules/` endpoin
   - Removed ElasticSearch from local API server code
   - Updated CollectionSearch to filter granule fields in addition to time frame for active collections
+- **CUMULUS-3847**
+  - list changes here
```
?

charleshuang80 replied:
placeholder, I'm still making sure I've captured everything. will make sure this is filled out when I'm fairly certain

npauzenga replied:
Cool. I'll leave this comment unresolved for now.

charleshuang80 replied:
this has been updated, and the line removed as suggested

npauzenga commented on packages/api/lambdas/cleanExecutions.js:
```diff
@@ -2,138 +2,143 @@
+// const moment = require('moment');
 const Logger = require('@cumulus/logger');
-const { sleep } = require('@cumulus/common');
+// const { sleep } = require('@cumulus/common');
```
Remove comments

npauzenga replied:
We should avoid having commented-out code generally. We can rely on the git diff if we need to reference what was here.

charleshuang80 replied:
hah. Well, I completely removed the lambda and test, and @jennyhliu suggested that I add them back and comment out except for a dummy handler. So I'm open to suggestions on why one method is better than the other...

charleshuang80 replied:
(see comment at top of file)

jennyhliu replied:
If we remove lambda and test here, we have to remove the related tf code, that's why I suggested to have dummy handler since we want to keep the lambda with RDS implementation.
Now I think we can delete the code and leave only the dummy handler.

charleshuang80 replied:
Ok, I can remove the commented code, if everyone is in agreement there. 
What about the test file, should I remove the file altogether? Or if not, what should I leave in the file if not commented out tests?

npauzenga replied:
Sounds good. Removing all commented code is certainly the preference as it's all in git if we need to get back to it.

charleshuang80 replied:
code removed but dummy handler left in place

npauzenga commented on packages/api/lib/testUtils.js:
```diff
@@ -620,20 +620,14 @@ const createAsyncOperationTestRecords = async (context) => {
   };
 };
 
+// this doesn't look like it's being used anywhere, should I remove it?
```
If it's not used, I'd say yes. It's an exported member so technically (possibly) a user-facing change for those using this package but... it's only test utils.

charleshuang80 replied:
removed as it was not used anywhere I could find

npauzenga commented on packages/async-operations/tests/test-async_operations.js:
```diff
@@ -61,7 +61,7 @@ test.before(async (t) => {
     Environment: {
       Variables: {
-        ES_HOST: 'es-host',
+        Timeout: 300,
```
What's this for?

charleshuang80 replied:
Since we're moving away from ES in general, ES indexing and all that, in the async operations tests I've changed all the places we're referencing ES operations to other options in those enums. Since we should probably be doing operations other than ES ones if ES is going away.

charleshuang80 replied:
So just as I changed the `operationType`s away from ES Index, I'm trying to move away from the `ES_HOST`. I wanted to keep a variable in here and found that Timeout could be one that is used.

npauzenga replied:
I see. And being that this is a mock, we just want to prove that we get back some semi-realistic config object.

npauzenga commented on packages/async-operations/tests/test-async_operations.js:
```diff
@@ -357,7 +357,7 @@ test('getLambdaEnvironmentVariables returns expected environment variables', (t)
 
   t.deepEqual(new Set(vars), new Set([
-    { name: 'ES_HOST', value: 'es-host' },
+    { name: 'Timeout', value: 300 },
```
Same as above, why this change?

charleshuang80 replied:
I answered this in the other thread, is that sufficient?

npauzenga commented on tf-modules/archive/reconciliation_report.tf:
```diff
@@ -15,8 +15,6 @@ resource "aws_lambda_function" "create_reconciliation_report" {
       CMR_HOST                         = var.cmr_custom_host
       DISTRIBUTION_ENDPOINT            = var.distribution_url
       ES_HOST                          = var.elasticsearch_hostname
```
What about ES_HOST?

charleshuang80 replied:
So I removed these two variables in reference to the comment at the very end of packages/api/lambdas/create-reconciliation-report.js, where I also removed them from logging.
I will be writing a ticket to remove ES from all the other terraform files where they're not needed (minus Cloud Metrics) so I thought this should be part of that removal.

charleshuang80 replied:
CUMULUS-3983

charleshuang80 commented on packages/api/lambdas/bulk-operation.js:
```diff
@@ -94,7 +94,7 @@ async function applyWorkflowToGranules({
  *   granule concurrency for the bulk deletion operation.  Defaults to 10
  * @param {Object} [payload.query] - Optional parameter of query to send to ES
- * @param {string} [payload.index] - Optional parameter of ES index to query.
+ * @param {string} [payload.index] - Optional parameter of ES index (metrics) to query.
```
Yes, my understanding of why we need to keep the ES related functionality is that for some of the granules and bulk operations it will still query Cloud Metrics for data. I don't feel like totally understand how it all ties together, but that is what is mentioned in code and what I was informed of. So I thought it would be helpful to clarify as much as possible that the ES indexing and querying is going to Cloud Metrics and that's why it is staying around and not being removed as part of ES removal.

npauzenga replied:
Ok, maybe just clarify "Cloud Metrics" or "Global Metrics" to make it clear what "metrics" means in this context?

npauzenga commented on CHANGELOG.md:
```diff
@@ -92,6 +92,12 @@ External tooling making use of `searchContext` in the `GET` `/granules/` endpoin
+  - for asyncOperations test data, change any ES related values to other options
+  - remove code from `@cumulus/api/lambdas/cleanExecutions` leaving a dummy handler, as the code worked with ES. lambda will be rewritten with CUMULUS-3982
+  - remove `@cumulus/api/endpoints/elasticsearch`, `@cumulus/api/lambdas/bootstrap`, and `@cumulus/api/lambdas/index-from-database`
+  - clarify in comments that ES functionality remaining in `@cumulus/api/lib/granules` is for working with CloudMetrics
```
Probably don't need this in the top-level release notes. The comments in the file (assuming they're there) are fine.

npauzenga commented on example/spec/serial/AsyncOperationRunnerFailingLambdaSpec.js:
```diff
@@ -49,7 +49,7 @@ describe('The AsyncOperation task runner executing a failing lambda function', (
         taskArn: randomString(),
         description: 'Some description',
-        operationType: 'ES Index',
+        operationType: 'Bulk Granules',
```
Just for clarity? Seems great, just curious.

charleshuang80 replied:
I'm sorry, I'm not sure I understand what you're asking here.

charleshuang80 replied:
yeah I got the types from /packages/types/api/async_operations. I switched it since we shouldn't be doing (most of) the ES things anymore. I considered removing the type, but Jonathan pointed out that past operations might still have the type. Not sure how long that will be for though.

npauzenga commented on packages/api/endpoints/elasticsearch.js:
```diff
@@ -1,233 +0,0 @@
-}
-
-// express routes
-router.put('/create-snapshot', createEsSnapshot);
```
Is there an argument for keeping any of this? Specifically the snapshot endpoint? Would this be useful for DAACs that still have an orphan ES instance?

charleshuang80 replied:
I don't know. If anything should be kept and we have a reason, that seems fine. But I don't know if any should be and I was just going with "remove all the ES things" except for what is needed for CloudMetrics and the other tickets I wrote. Please let me know if there's anything that should be put back.

charleshuang80 replied:
the functionality for the endpoint was never actually implemented, so keeping this would not do anything.

npauzenga commented on packages/api/lambdas/create-reconciliation-report.js:
```diff
@@ -900,9 +895,7 @@ async function processRequest(params) {
-  let reportApiRecord = translatePostgresReconReportToApiReconReport(reportPgRecord);
-  await indexReconciliationReport(esClient, reportApiRecord, process.env.ES_INDEX);
-  log.info(`Report added to database as pending: ${JSON.stringify(reportApiRecord)}.`);
+  log.info(`Report added to database as pending: ${JSON.stringify(reportPgRecord)}.`);
```
Is there no value in logging the API-formatted record as we did previously? Generally users don't see the pg record, they get the translated version. But if we're just translating to log it... that may be overkill.

charleshuang80 replied:
So yeah, the way this is written, we are translating it at the very end to return the api format record, but this seems to be the only place there was explicit logging. I'm not sure for logging purposes if api format is better than the pg format, so I'm open to doing it that way. But I did just log what we had instead of translating it for logging.
Also, thinking about it more, I'm open to adding some more logging like maybe at the end before returning the api format record. What do you think?

npauzenga replied:
If it doesn't make sense to move this log statement or translated beforehand, maybe just update the message to make it clear that it's a PG-formatted record. It's a nit. I'm just concerned that someone sees this in the logs and gets confused as to why the naming conventions are different. More logging is good.

charleshuang80 replied:
for consistency, it seems correct to keep the translation and logging of the api format. the translation is a quick function anyways. Added a comment for future reference because it might be confusing why the translation was done just for logging.
Also added a bit more logging (with translation)

npauzenga commented on packages/api/lib/granules.js:
```diff
@@ -234,10 +234,10 @@ function getTotalHits(bodyHits) {
  *
  * @param {Object} payload
- * @param {string} [payload.index] - ES index to query
+ * @param {string} [payload.index] - ES (metrics) index to query
```
Same comment as above. Maybe clarify what 'metrics' means in this context.

charleshuang80 replied:
this was done in a few places.

npauzenga commented on packages/api/lib/testUtils.js:
```diff
@@ -620,22 +617,6 @@ const createAsyncOperationTestRecords = async (context) => {
-    esIndex,
-  } = context;
-
-  await executionPgModel.delete(knex, { arn });
```
Don't we need this?

charleshuang80 replied:
Hah, I mentioned in a previous comment that I did a search for the method `cleanupExecutionTestRecords` and did not find it being used anywhere. So I wondered if I should just remove it as I couldn't find it being used, and you confirmed...

npauzenga replied:
Ah ok, sure. Didn't piece together that this was the function we weren't sure was used.

npauzenga reviewed (changes requested):
Had a couple comments. Mostly clarifying questions.

Are there any tests failing/blockers in CI?

charleshuang80 commented:
> Had a couple comments. Mostly clarifying questions.
> 
> Are there any tests failing/blockers in CI?

Yes, I posted about an aws-client test I'm having issues with in slack.

charleshuang80 commented on packages/api/tests/lambdas/test-cleanExecutions.js:
removed as per convo during PI planning

npauzenga reviewed (changes requested):
Thanks @charleshuang80! This is looking great. I think (pending anything CI uncovers) this should be good to go. I searched the repo and found a couple areas I'm curious about. Notably:
- https://github.com/nasa/cumulus/blob/CUMULUS-3847/packages/api/tests/performance/lib/test-write-granules.js#L75-L83

(And other tests that still import es-client) Should those be updated or is that scoped somewhere else?

charleshuang80 commented:
> Thanks @charleshuang80! This is looking great. I think (pending anything CI uncovers) this should be good to go. I searched the repo and found a couple areas I'm curious about. Notably:
> 
> * https://github.com/nasa/cumulus/blob/CUMULUS-3847/packages/api/tests/performance/lib/test-write-granules.js#L75-L83
> 
> (And other tests that still import es-client) Should those be updated or is that scoped somewhere else?

I think it depends on the test. A few that utilize the cloud metrics functionality still need it I believe. But this test failed in CI and I've been working on removing the ES stuff and trying to figure out if we can still test the performance. I'll do another search, but if you have other specific examples you think should go or just aren't sure about please let me know.

charleshuang80 commented:
> Thanks @charleshuang80! This is looking great. I think (pending anything CI uncovers) this should be good to go. I searched the repo and found a couple areas I'm curious about. Notably:
> 
> * https://github.com/nasa/cumulus/blob/CUMULUS-3847/packages/api/tests/performance/lib/test-write-granules.js#L75-L83
> 
> (And other tests that still import es-client) Should those be updated or is that scoped somewhere else?

So I fixed this file and tested it. Also found a few more references that I removed and tested in the latest commit. There are some references in like granules, test-executions, test-build-operation, and test-granules that I think need to stay for Cloud Metrics. If there are any other locations you think I missed please let me know.

npauzenga reviewed (approved):
Thanks @charleshuang80! I think you've got everything that jumps out. If there are areas that still have some ES cruft, we can remove as we find it. The important thing is that the tests pass and all of the core functionality still... functions 😄
